core/shared/src/main/scala/org/http4s/UriTemplate.scala

```diff
@@ -117,7 +116,9 @@ object UriTemplate {
   def isUnreserved(s: String): Boolean = s.forall(unreserved.contains)
 
   def isUnreservedOrEncoded(s: String): Boolean =
-    URLEncoder.encode(s, "UTF-8").forall(c => unreserved.contains(c) || c == '%')
+    Uri
+      .encode(s, spaceIsPlus = true, toSkip = unreserved)
+      .forall(c => unreserved.contains(c) || c == '%')
```

Comment:
I'd appreciate some eyes on this (`URLEncoder` is JVM-only). Also, can somebody explain what this method is doing anyway? Why encode a string, then check if it is encoded?